apps/wallet-dashboard/components/Dialogs/Staking/StakeDialog.tsx

```diff
@@ -107,7 +107,12 @@ export function StakeDialog({
     }
 
     function detailsHandleStake() {
-        setView?.(StakeDialogView.SelectValidator);
+        setSelectedValidator?.(stakedDetails?.validatorAddress ?? '');
+        setView?.(
+            isTimelockedStaking
+                ? StakeDialogView.EnterTimelockedAmount
+                : StakeDialogView.EnterAmount,
+        );
```

Comment:
The same code is used in both functions. Also, since `?? ''` has been added to `setSelectedValidator` before, I think it's a good idea to check if a validator exists before switching to the view
```suggestion
        selectValidatorHandleNext()
```